stubs/synapse/synapse_rust.pyi

```diff
@@ -0,0 +1 @@
+def sum_as_string(a: int, b: int) -> str: ...
```

Comment:
Is there a way to automatically generate this (in the future)?

Also: Python's `int`s are bigger than Rust's `usize`s. What happens if we call `sum_as_string(2**100, 2**200)` with bigger-than-usize ints?

Reply:
Also: does `stubs/synapse/__init__.pyi` clobber the annotations in `synapse/__init__.py`? Oh, that file is empty except for a copyright notice.

Reply:
> Is there a way to automatically generate this (in the future)?

I don't believe anyone has written any tooling for it, annoyingly.

> Also: Python's ints are bigger than Rust's usizes. What happens if we call sum_as_string(2**100, 2**200) with bigger-than-usize ints?

It will raise an `OverflowError`. In general `pyo3` will raise an exception whenever it tries to convert from a python to rust type.

> Also: does stubs/synapse/__init__.pyi clobber the annotations in synapse/__init__.py? Oh, that file is empty except for a copyright notice.

Empirically `mypy` does the right thing here and uses `synapse/__init__.py` (I just tested it by doing `reveal_type(synapse.__version__) `